```
Error running mint automation
+ mkdir -p 12342-b2a884d/gopath/src/github.com/minio
+ git -C 12342-b2a884d/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 12342-b2a884d/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 12342-b2a884d/gopath/src/github.com/minio/minio fetch --quiet minio pull/12342/head:pr12342
+ git -C 12342-b2a884d/gopath/src/github.com/minio/minio checkout --quiet pr12342
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/12342-b2a884d/gopath
+ make -C 12342-b2a884d/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/minio/cli v1.22.0
go: downloading cloud.google.com/go v0.65.0
go: downloading github.com/minio/madmin-go v1.0.2
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading google.golang.org/api v0.31.0
go: downloading github.com/json-iterator/go v1.1.11
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/minio/sio v0.3.0
go: downloading golang.org/x/sys v0.0.0-20210510120138-977fb7262007
go: downloading github.com/fatih/color v1.10.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/minio/kes v0.14.0
go: downloading golang.org/x/crypto v0.0.0-20210421170649-83a5a9bb288b
go: downloading github.com/VividCortex/ewma v1.1.1
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/klauspost/compress v1.12.2
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/bits-and-blooms/bloom/v3 v3.0.1
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/minio/rpc v1.0.0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/shirou/gopsutil/v3 v3.21.3
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/miekg/dns v1.1.35
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b
go: downloading github.com/pierrec/lz4 v2.6.0+incompatible
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/tinylib/msgp v1.1.6-0.20210521143832-0becd170c402
go: downloading github.com/go-ldap/ldap v3.0.2+incompatible
go: downloading github.com/dswarbrick/smart v0.0.0-20190505152634-909a45200d6d
go: downloading github.com/beevik/ntp v0.3.0
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading github.com/elastic/go-elasticsearch/v7 v7.12.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/lib/pq v1.9.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/nats-io/stan.go v0.8.3
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/minio/parquet-go v1.0.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/mattn/go-runewidth v0.0.7
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading golang.org/x/net v0.0.0-20210421230115-4e50805a0758
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/bits-and-blooms/bitset v1.2.0
go: downloading github.com/golang/snappy v0.0.3
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/tidwall/sjson v1.1.6
go: downloading github.com/tidwall/gjson v1.7.5
go: downloading golang.org/x/text v0.3.6
go: downloading github.com/nats-io/nkeys v0.2.0
go: downloading github.com/nats-io/jwt v1.1.0
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/minio/argon2 v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading go.uber.org/atomic v1.6.0
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/tidwall/pretty v1.1.0
go: downloading github.com/tidwall/match v1.0.3
go: downloading github.com/tklauser/go-sysconf v0.3.4
go: downloading cloud.google.com/go/storage v1.10.0
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/tklauser/numcpus v0.2.1
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/rs/xid v1.2.1
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading google.golang.org/grpc v1.29.1
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/klauspost/cpuid/v2 v2.0.4
go: downloading google.golang.org/genproto v0.0.0-20200901141002-b3bf27a9dbd1
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: downloading github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
go: downloading github.com/josharian/intern v1.0.0
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
# github.com/minio/minio/cmd
cmd/bucket-lifecycle.go:223:4: unknown field 'RemoteVersionID' in struct literal of type jentry
cmd/bucket-lifecycle.go:338:78: oi.transitionRemoteVersionID undefined (type ObjectInfo has no field or method transitionRemoteVersionID)
cmd/erasure-metadata.go:155:9: objInfo.transitionRemoteVersionID undefined (type ObjectInfo has no field or method transitionRemoteVersionID)
cmd/erasure-metadata.go:155:40: fi.TransitionRemoteVersionID undefined (type FileInfo has no field or method TransitionRemoteVersionID)
cmd/erasure-object.go:1341:4: fi.TransitionRemoteVersionID undefined (type FileInfo has no field or method TransitionRemoteVersionID)
cmd/storage-datatypes_gen.go:597:3: z.TransitionRemoteVersionID undefined (type *FileInfo has no field or method TransitionRemoteVersionID)
cmd/storage-datatypes_gen.go:768:24: z.TransitionRemoteVersionID undefined (type *FileInfo has no field or method TransitionRemoteVersionID)
cmd/storage-datatypes_gen.go:883:28: z.TransitionRemoteVersionID undefined (type *FileInfo has no field or method TransitionRemoteVersionID)
cmd/storage-datatypes_gen.go:969:3: z.TransitionRemoteVersionID undefined (type *FileInfo has no field or method TransitionRemoteVersionID)
cmd/storage-datatypes_gen.go:1096: z.TransitionRemoteVersionID undefined (type *FileInfo has no field or method TransitionRemoteVersionID)
cmd/storage-datatypes_gen.go:1096: too many errors
make: *** [build] Error 2

```

--- a/cmd/bucket-lifecycle.go
+++ b/cmd/bucket-lifecycle.go
@@ -46,6 +46,8 @@ const (
 	TransitionStatus = "transition-status"
 	// TransitionedObjectName name of transitioned object
 	TransitionedObjectName = "transitioned-object"
+	// TransitionedRemoteVersionID is version of remote object
+	TransitionedRemoteVersionID = "transitioned-remote-ver-id"
 	// TransitionTier name of transition storage class
 	TransitionTier = "transition-tier"
 )

--- a/cmd/bucket-lifecycle.go
+++ b/cmd/bucket-lifecycle.go
@@ -46,8 +46,8 @@ const (
 	TransitionStatus = "transition-status"
 	// TransitionedObjectName name of transitioned object
 	TransitionedObjectName = "transitioned-object"
-	// TransitionedRemoteVersionID is version of remote object
-	TransitionedRemoteVersionID = "transitioned-remote-ver-id"
+	// TransitionedVersionID is version of remote object
+	TransitionedVersionID = "transitioned-versionID"
 	// TransitionTier name of transition storage class
 	TransitionTier = "transition-tier"
 )
@@ -219,9 +219,9 @@ func expireTransitionedObject(ctx context.Context, objectAPI ObjectLayer, oi *Ob
 		// When an object is past expiry or when a transitioned object is being
 		// deleted, 'mark' the data in the remote tier for delete.
 		entry := jentry{
-			ObjName:         oi.transitionedObjName,
-			RemoteVersionID: oi.transitionRemoteVersionID,
-			TierName:        oi.TransitionTier,
+			ObjName:   oi.transitionedObjName,
+			VersionID: oi.transitionVersionID,
+			TierName:  oi.TransitionTier,
 		}
 		if err := globalTierJournal.AddEntry(entry); err != nil {
 			logger.LogIf(ctx, err)
@@ -335,7 +335,7 @@ func getTransitionedObjectReader(ctx context.Context, bucket, object string, rs 
 		gopts.length = length
 	}
 
-	reader, err := tgtClient.Get(ctx, oi.transitionedObjName, remoteVersionID(oi.transitionRemoteVersionID), gopts)
+	reader, err := tgtClient.Get(ctx, oi.transitionedObjName, remoteVersionID(oi.transitionVersionID), gopts)
 	if err != nil {
 		return nil, err
 	}

--- a/cmd/bucket-lifecycle.go
+++ b/cmd/bucket-lifecycle.go
@@ -208,41 +208,46 @@ const (
 //
 // 1. when a restored (via PostRestoreObject API) object expires.
 // 2. when a transitioned object expires (based on an ILM rule).
-func expireTransitionedObject(ctx context.Context, objectAPI ObjectLayer, bucket, object string, lcOpts lifecycle.ObjectOpts, remoteObject, tier string, action expireAction) error {
+func expireTransitionedObject(ctx context.Context, objectAPI ObjectLayer, oi *ObjectInfo, lcOpts lifecycle.ObjectOpts, action expireAction) error {
 	var opts ObjectOptions
-	opts.Versioned = globalBucketVersioningSys.Enabled(bucket)
+	opts.Versioned = globalBucketVersioningSys.Enabled(oi.Bucket)
 	opts.VersionID = lcOpts.VersionID
 	switch action {
 	case expireObj:
 		// When an object is past expiry or when a transitioned object is being
 		// deleted, 'mark' the data in the remote tier for delete.
-		if err := globalTierJournal.AddEntry(jentry{ObjName: remoteObject, TierName: tier}); err != nil {
+		entry := jentry{
+			ObjName:         oi.transitionedObjName,
+			RemoteVersionID: oi.transitionRemoteVersionID,
+			TierName:        oi.TransitionTier,
+		}
+		if err := globalTierJournal.AddEntry(entry); err != nil {
 			logger.LogIf(ctx, err)
 			return err
 		}
 		// Delete metadata on source, now that data in remote tier has been
 		// marked for deletion.
-		if _, err := objectAPI.DeleteObject(ctx, bucket, object, opts); err != nil {
+		if _, err := objectAPI.DeleteObject(ctx, oi.Bucket, oi.Name, opts); err != nil {
 			logger.LogIf(ctx, err)
 			return err
 		}
 
 		// Send audit for the lifecycle delete operation
-		auditLogLifecycle(ctx, bucket, object)
+		auditLogLifecycle(ctx, oi.Bucket, oi.Name)
 
 		eventName := event.ObjectRemovedDelete
 		if lcOpts.DeleteMarker {
 			eventName = event.ObjectRemovedDeleteMarkerCreated
 		}
 		objInfo := ObjectInfo{
-			Name:         object,
+			Name:         oi.Name,
 			VersionID:    lcOpts.VersionID,
 			DeleteMarker: lcOpts.DeleteMarker,
 		}
 		// Notify object deleted event.
 		sendEvent(eventArgs{
 			EventName:  eventName,
-			BucketName: bucket,
+			BucketName: oi.Bucket,
 			Object:     objInfo,
 			Host:       "Internal: [ILM-EXPIRY]",
 		})
@@ -252,7 +257,7 @@ func expireTransitionedObject(ctx context.Context, objectAPI ObjectLayer, bucket
 		// from the source, while leaving metadata behind. The data on
 		// transitioned tier lies untouched and still accessible
 		opts.Transition.ExpireRestored = true
-		_, err := objectAPI.DeleteObject(ctx, bucket, object, opts)
+		_, err := objectAPI.DeleteObject(ctx, oi.Bucket, oi.Name, opts)
 		return err
 	default:
 		return fmt.Errorf("Unknown expire action %v", action)
@@ -286,11 +291,12 @@ func transitionObject(ctx context.Context, objectAPI ObjectLayer, oi ObjectInfo)
 		UserTags: oi.UserTags,
 	}
 	tierName := getLifeCycleTransitionTier(ctx, lc, oi.Bucket, lcOpts)
-	opts := ObjectOptions{Transition: TransitionOptions{
-		Status: lifecycle.TransitionPending,
-		Tier:   tierName,
-		ETag:   oi.ETag,
-	},
+	opts := ObjectOptions{
+		Transition: TransitionOptions{
+			Status: lifecycle.TransitionPending,
+			Tier:   tierName,
+			ETag:   oi.ETag,
+		},
 		VersionID: oi.VersionID,
 		Versioned: globalBucketVersioningSys.Enabled(oi.Bucket),
 		MTime:     oi.ModTime,
@@ -327,7 +333,7 @@ func getTransitionedObjectReader(ctx context.Context, bucket, object string, rs 
 		gopts.length = length
 	}
 
-	reader, err := tgtClient.Get(ctx, oi.transitionedObjName, gopts)
+	reader, err := tgtClient.Get(ctx, oi.transitionedObjName, remoteVersionID(oi.transitionRemoteVersionID), gopts)
 	if err != nil {
 		return nil, err
 	}

--- a/cmd/data-scanner.go
+++ b/cmd/data-scanner.go
@@ -1062,7 +1062,7 @@ func applyExpiryOnTransitionedObject(ctx context.Context, objLayer ObjectLayer, 
 	if restoredObject {
 		action = expireRestoredObj
 	}
-	if err := expireTransitionedObject(ctx, objLayer, obj.Bucket, obj.Name, lcOpts, obj.transitionedObjName, obj.TransitionTier, action); err != nil {
+	if err := expireTransitionedObject(ctx, objLayer, &obj, lcOpts, action); err != nil {
 		if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 			return false
 		}

--- a/cmd/erasure-metadata.go
+++ b/cmd/erasure-metadata.go
@@ -155,6 +155,7 @@ func (fi FileInfo) ToObjectInfo(bucket, object string) ObjectInfo {
 
 	objInfo.TransitionStatus = fi.TransitionStatus
 	objInfo.transitionedObjName = fi.TransitionedObjName
+	objInfo.transitionRemoteVersionID = fi.TransitionRemoteVersionID
 	objInfo.TransitionTier = fi.TransitionTier
 
 	// etag/md5Sum has already been extracted. We need to

--- a/cmd/erasure-metadata.go
+++ b/cmd/erasure-metadata.go
@@ -155,7 +155,7 @@ func (fi FileInfo) ToObjectInfo(bucket, object string) ObjectInfo {
 
 	objInfo.TransitionStatus = fi.TransitionStatus
 	objInfo.transitionedObjName = fi.TransitionedObjName
-	objInfo.transitionRemoteVersionID = fi.TransitionRemoteVersionID
+	objInfo.transitionVersionID = fi.TransitionVersionID
 	objInfo.TransitionTier = fi.TransitionTier
 
 	// etag/md5Sum has already been extracted. We need to

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -1358,7 +1358,7 @@ func (er erasureObjects) TransitionObject(ctx context.Context, bucket, object st
 	fi.TransitionStatus = lifecycle.TransitionComplete
 	fi.TransitionedObjName = destObj
 	fi.TransitionTier = opts.Transition.Tier
-	fi.TransitionRemoteVersionID = string(rv)
+	fi.TransitionVersionID = string(rv)
 	eventName := event.ObjectTransitionComplete
 
 	storageDisks := er.getDisks()

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -1348,7 +1348,8 @@ func (er erasureObjects) TransitionObject(ctx context.Context, bucket, object st
 		pw.CloseWithError(err)
 	}()
 
-	err = tgtClient.Put(ctx, destObj, pr, fi.Size)
+	var rv remoteVersionID
+	rv, err = tgtClient.Put(ctx, destObj, pr, fi.Size)
 	pr.CloseWithError(err)
 	if err != nil {
 		logger.LogIf(ctx, fmt.Errorf("Unable to transition %s/%s(%s) to %s tier: %w", bucket, object, opts.VersionID, opts.Transition.Tier, err))
@@ -1357,6 +1358,7 @@ func (er erasureObjects) TransitionObject(ctx context.Context, bucket, object st
 	fi.TransitionStatus = lifecycle.TransitionComplete
 	fi.TransitionedObjName = destObj
 	fi.TransitionTier = opts.Transition.Tier
+	fi.TransitionRemoteVersionID = string(rv)
 	eventName := event.ObjectTransitionComplete
 
 	storageDisks := er.getDisks()

--- a/cmd/object-api-datatypes.go
+++ b/cmd/object-api-datatypes.go
@@ -117,6 +117,8 @@ type ObjectInfo struct {
 	TransitionStatus string
 	// Name of transitioned object on remote tier
 	transitionedObjName string
+	// VersionID on the the remote tier
+	transitionRemoteVersionID string
 	// Name of remote tier object has transitioned to
 	TransitionTier string
 

--- a/cmd/object-api-datatypes.go
+++ b/cmd/object-api-datatypes.go
@@ -118,7 +118,7 @@ type ObjectInfo struct {
 	// Name of transitioned object on remote tier
 	transitionedObjName string
 	// VersionID on the the remote tier
-	transitionRemoteVersionID string
+	transitionVersionID string
 	// Name of remote tier object has transitioned to
 	TransitionTier string
 

--- a/cmd/storage-datatypes.go
+++ b/cmd/storage-datatypes.go
@@ -138,6 +138,9 @@ type FileInfo struct {
 	TransitionedObjName string
 	// TransitionTier is the storage class label assigned to remote tier.
 	TransitionTier string
+	// TransitionRemoteVersionID stores a version ID of the object associate
+	// with the remote tier.
+	TransitionRemoteVersionID string
 	// ExpireRestored indicates that the restored object is to be expired.
 	ExpireRestored bool
 

--- a/cmd/storage-datatypes.go
+++ b/cmd/storage-datatypes.go
@@ -138,9 +138,9 @@ type FileInfo struct {
 	TransitionedObjName string
 	// TransitionTier is the storage class label assigned to remote tier.
 	TransitionTier string
-	// TransitionRemoteVersionID stores a version ID of the object associate
+	// TransitionVersionID stores a version ID of the object associate
 	// with the remote tier.
-	TransitionRemoteVersionID string
+	TransitionVersionID string
 	// ExpireRestored indicates that the restored object is to be expired.
 	ExpireRestored bool
 

--- a/cmd/storage-datatypes_gen.go
+++ b/cmd/storage-datatypes_gen.go
@@ -594,9 +594,9 @@ func (z *FileInfo) DecodeMsg(dc *msgp.Reader) (err error) {
 		err = msgp.WrapError(err, "TransitionTier")
 		return
 	}
-	z.TransitionRemoteVersionID, err = dc.ReadString()
+	z.TransitionVersionID, err = dc.ReadString()
 	if err != nil {
-		err = msgp.WrapError(err, "TransitionRemoteVersionID")
+		err = msgp.WrapError(err, "TransitionVersionID")
 		return
 	}
 	z.ExpireRestored, err = dc.ReadBool()
@@ -765,9 +765,9 @@ func (z *FileInfo) EncodeMsg(en *msgp.Writer) (err error) {
 		err = msgp.WrapError(err, "TransitionTier")
 		return
 	}
-	err = en.WriteString(z.TransitionRemoteVersionID)
+	err = en.WriteString(z.TransitionVersionID)
 	if err != nil {
-		err = msgp.WrapError(err, "TransitionRemoteVersionID")
+		err = msgp.WrapError(err, "TransitionVersionID")
 		return
 	}
 	err = en.WriteBool(z.ExpireRestored)
@@ -880,7 +880,7 @@ func (z *FileInfo) MarshalMsg(b []byte) (o []byte, err error) {
 	o = msgp.AppendString(o, z.TransitionStatus)
 	o = msgp.AppendString(o, z.TransitionedObjName)
 	o = msgp.AppendString(o, z.TransitionTier)
-	o = msgp.AppendString(o, z.TransitionRemoteVersionID)
+	o = msgp.AppendString(o, z.TransitionVersionID)
 	o = msgp.AppendBool(o, z.ExpireRestored)
 	o = msgp.AppendString(o, z.DataDir)
 	o = msgp.AppendBool(o, z.XLV1)
@@ -966,9 +966,9 @@ func (z *FileInfo) UnmarshalMsg(bts []byte) (o []byte, err error) {
 		err = msgp.WrapError(err, "TransitionTier")
 		return
 	}
-	z.TransitionRemoteVersionID, bts, err = msgp.ReadStringBytes(bts)
+	z.TransitionVersionID, bts, err = msgp.ReadStringBytes(bts)
 	if err != nil {
-		err = msgp.WrapError(err, "TransitionRemoteVersionID")
+		err = msgp.WrapError(err, "TransitionVersionID")
 		return
 	}
 	z.ExpireRestored, bts, err = msgp.ReadBoolBytes(bts)
@@ -1093,7 +1093,7 @@ func (z *FileInfo) UnmarshalMsg(bts []byte) (o []byte, err error) {
 
 // Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
 func (z *FileInfo) Msgsize() (s int) {
-	s = 3 + msgp.StringPrefixSize + len(z.Volume) + msgp.StringPrefixSize + len(z.Name) + msgp.StringPrefixSize + len(z.VersionID) + msgp.BoolSize + msgp.BoolSize + msgp.StringPrefixSize + len(z.TransitionStatus) + msgp.StringPrefixSize + len(z.TransitionedObjName) + msgp.StringPrefixSize + len(z.TransitionTier) + msgp.StringPrefixSize + len(z.TransitionRemoteVersionID) + msgp.BoolSize + msgp.StringPrefixSize + len(z.DataDir) + msgp.BoolSize + msgp.TimeSize + msgp.Int64Size + msgp.Uint32Size + msgp.MapHeaderSize
+	s = 3 + msgp.StringPrefixSize + len(z.Volume) + msgp.StringPrefixSize + len(z.Name) + msgp.StringPrefixSize + len(z.VersionID) + msgp.BoolSize + msgp.BoolSize + msgp.StringPrefixSize + len(z.TransitionStatus) + msgp.StringPrefixSize + len(z.TransitionedObjName) + msgp.StringPrefixSize + len(z.TransitionTier) + msgp.StringPrefixSize + len(z.TransitionVersionID) + msgp.BoolSize + msgp.StringPrefixSize + len(z.DataDir) + msgp.BoolSize + msgp.TimeSize + msgp.Int64Size + msgp.Uint32Size + msgp.MapHeaderSize
 	if z.Metadata != nil {
 		for za0001, za0002 := range z.Metadata {
 			_ = za0002

--- a/cmd/storage-datatypes_gen.go
+++ b/cmd/storage-datatypes_gen.go
@@ -550,8 +550,8 @@ func (z *FileInfo) DecodeMsg(dc *msgp.Reader) (err error) {
 		err = msgp.WrapError(err)
 		return
 	}
-	if zb0001 != 23 {
-		err = msgp.ArrayError{Wanted: 23, Got: zb0001}
+	if zb0001 != 24 {
+		err = msgp.ArrayError{Wanted: 24, Got: zb0001}
 		return
 	}
 	z.Volume, err = dc.ReadString()
@@ -592,6 +592,11 @@ func (z *FileInfo) DecodeMsg(dc *msgp.Reader) (err error) {
 	z.TransitionTier, err = dc.ReadString()
 	if err != nil {
 		err = msgp.WrapError(err, "TransitionTier")
+		return
+	}
+	z.TransitionRemoteVersionID, err = dc.ReadString()
+	if err != nil {
+		err = msgp.WrapError(err, "TransitionRemoteVersionID")
 		return
 	}
 	z.ExpireRestored, err = dc.ReadBool()
@@ -715,8 +720,8 @@ func (z *FileInfo) DecodeMsg(dc *msgp.Reader) (err error) {
 
 // EncodeMsg implements msgp.Encodable
 func (z *FileInfo) EncodeMsg(en *msgp.Writer) (err error) {
-	// array header, size 23
-	err = en.Append(0xdc, 0x0, 0x17)
+	// array header, size 24
+	err = en.Append(0xdc, 0x0, 0x18)
 	if err != nil {
 		return
 	}
@@ -758,6 +763,11 @@ func (z *FileInfo) EncodeMsg(en *msgp.Writer) (err error) {
 	err = en.WriteString(z.TransitionTier)
 	if err != nil {
 		err = msgp.WrapError(err, "TransitionTier")
+		return
+	}
+	err = en.WriteString(z.TransitionRemoteVersionID)
+	if err != nil {
+		err = msgp.WrapError(err, "TransitionRemoteVersionID")
 		return
 	}
 	err = en.WriteBool(z.ExpireRestored)
@@ -860,8 +870,8 @@ func (z *FileInfo) EncodeMsg(en *msgp.Writer) (err error) {
 // MarshalMsg implements msgp.Marshaler
 func (z *FileInfo) MarshalMsg(b []byte) (o []byte, err error) {
 	o = msgp.Require(b, z.Msgsize())
-	// array header, size 23
-	o = append(o, 0xdc, 0x0, 0x17)
+	// array header, size 24
+	o = append(o, 0xdc, 0x0, 0x18)
 	o = msgp.AppendString(o, z.Volume)
 	o = msgp.AppendString(o, z.Name)
 	o = msgp.AppendString(o, z.VersionID)
@@ -870,6 +880,7 @@ func (z *FileInfo) MarshalMsg(b []byte) (o []byte, err error) {
 	o = msgp.AppendString(o, z.TransitionStatus)
 	o = msgp.AppendString(o, z.TransitionedObjName)
 	o = msgp.AppendString(o, z.TransitionTier)
+	o = msgp.AppendString(o, z.TransitionRemoteVersionID)
 	o = msgp.AppendBool(o, z.ExpireRestored)
 	o = msgp.AppendString(o, z.DataDir)
 	o = msgp.AppendBool(o, z.XLV1)
@@ -911,8 +922,8 @@ func (z *FileInfo) UnmarshalMsg(bts []byte) (o []byte, err error) {
 		err = msgp.WrapError(err)
 		return
 	}
-	if zb0001 != 23 {
-		err = msgp.ArrayError{Wanted: 23, Got: zb0001}
+	if zb0001 != 24 {
+		err = msgp.ArrayError{Wanted: 24, Got: zb0001}
 		return
 	}
 	z.Volume, bts, err = msgp.ReadStringBytes(bts)
@@ -953,6 +964,11 @@ func (z *FileInfo) UnmarshalMsg(bts []byte) (o []byte, err error) {
 	z.TransitionTier, bts, err = msgp.ReadStringBytes(bts)
 	if err != nil {
 		err = msgp.WrapError(err, "TransitionTier")
+		return
+	}
+	z.TransitionRemoteVersionID, bts, err = msgp.ReadStringBytes(bts)
+	if err != nil {
+		err = msgp.WrapError(err, "TransitionRemoteVersionID")
 		return
 	}
 	z.ExpireRestored, bts, err = msgp.ReadBoolBytes(bts)
@@ -1077,7 +1093,7 @@ func (z *FileInfo) UnmarshalMsg(bts []byte) (o []byte, err error) {
 
 // Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
 func (z *FileInfo) Msgsize() (s int) {
-	s = 3 + msgp.StringPrefixSize + len(z.Volume) + msgp.StringPrefixSize + len(z.Name) + msgp.StringPrefixSize + len(z.VersionID) + msgp.BoolSize + msgp.BoolSize + msgp.StringPrefixSize + len(z.TransitionStatus) + msgp.StringPrefixSize + len(z.TransitionedObjName) + msgp.StringPrefixSize + len(z.TransitionTier) + msgp.BoolSize + msgp.StringPrefixSize + len(z.DataDir) + msgp.BoolSize + msgp.TimeSize + msgp.Int64Size + msgp.Uint32Size + msgp.MapHeaderSize
+	s = 3 + msgp.StringPrefixSize + len(z.Volume) + msgp.StringPrefixSize + len(z.Name) + msgp.StringPrefixSize + len(z.VersionID) + msgp.BoolSize + msgp.BoolSize + msgp.StringPrefixSize + len(z.TransitionStatus) + msgp.StringPrefixSize + len(z.TransitionedObjName) + msgp.StringPrefixSize + len(z.TransitionTier) + msgp.StringPrefixSize + len(z.TransitionRemoteVersionID) + msgp.BoolSize + msgp.StringPrefixSize + len(z.DataDir) + msgp.BoolSize + msgp.TimeSize + msgp.Int64Size + msgp.Uint32Size + msgp.MapHeaderSize
 	if z.Metadata != nil {
 		for za0001, za0002 := range z.Metadata {
 			_ = za0002

--- a/cmd/storage-rest-common.go
+++ b/cmd/storage-rest-common.go
@@ -18,7 +18,7 @@
 package cmd
 
 const (
-	storageRESTVersion       = "v35" // Inline bugfix needs all servers to be updated
+	storageRESTVersion       = "v36" // Changes to FileInfo for tier-journal
 	storageRESTVersionPrefix = SlashSeparator + storageRESTVersion
 	storageRESTPrefix        = minioReservedBucketPath + "/storage"
 )

--- a/cmd/tier-journal.go
+++ b/cmd/tier-journal.go
@@ -42,13 +42,18 @@ type tierJournal struct {
 }
 
 type jentry struct {
-	ObjName  string `msg:"obj"`
-	TierName string `msg:"tier"`
+	ObjName         string `msg:"obj"`
+	RemoteVersionID string `msg:"rvid"`
+	TierName        string `msg:"tier"`
 }
 
 const (
 	tierJournalVersion = 1
 	tierJournalHdrLen  = 2 // 2 bytes
+)
+
+var (
+	errUnsupportedJournalVersion = errors.New("unsupported pending deletes journal version")
 )
 
 func initTierDeletionJournal(done <-chan struct{}) (*tierJournal, error) {
@@ -84,7 +89,7 @@ func (j *tierJournal) rotate() error {
 	return j.Open()
 }
 
-type walkFn func(objName, tierName string) error
+type walkFn func(objName, rvID, tierName string) error
 
 func (j *tierJournal) ReadOnlyPath() string {
 	return filepath.Join(j.diskPath, minioMetaBucket, "ilm", "deletion-journal.ro.bin")
@@ -111,6 +116,7 @@ func (j *tierJournal) WalkEntries(fn walkFn) {
 	}
 	defer ro.Close()
 	mr := msgp.NewReader(ro)
+
 	done := false
 	for {
 		var entry jentry
@@ -123,9 +129,11 @@ func (j *tierJournal) WalkEntries(fn walkFn) {
 			logger.LogIf(context.Background(), fmt.Errorf("tier-journal: failed to decode journal entry %s", err))
 			break
 		}
-		err = fn(entry.ObjName, entry.TierName)
+		err = fn(entry.ObjName, entry.RemoteVersionID, entry.TierName)
 		if err != nil && !isErrObjectNotFound(err) {
 			logger.LogIf(context.Background(), fmt.Errorf("tier-journal: failed to delete transitioned object %s from %s due to %s", entry.ObjName, entry.TierName, err))
+			// We add the entry into the active journal to try again
+			// later.
 			j.AddEntry(entry)
 		}
 	}
@@ -134,12 +142,12 @@ func (j *tierJournal) WalkEntries(fn walkFn) {
 	}
 }
 
-func deleteObjectFromRemoteTier(objName, tierName string) error {
+func deleteObjectFromRemoteTier(objName, rvID, tierName string) error {
 	w, err := globalTierConfigMgr.getDriver(tierName)
 	if err != nil {
 		return err
 	}
-	err = w.Remove(context.Background(), objName)
+	err = w.Remove(context.Background(), objName, remoteVersionID(rvID))
 	if err != nil {
 		return err
 	}
@@ -263,8 +271,15 @@ func (j *tierJournal) OpenRO() (io.ReadCloser, error) {
 
 	switch binary.LittleEndian.Uint16(data[:]) {
 	case tierJournalVersion:
+		return file, nil
 	default:
-		return nil, errors.New("unsupported pending deletes journal version")
+		return nil, errUnsupportedJournalVersion
 	}
-	return file, nil
+}
+
+// jentryV1 represents the entry in the journal before RemoteVersionID was
+// added. It remains here for use in tests for the struct element addition.
+type jentryV1 struct {
+	ObjName  string `msg:"obj"`
+	TierName string `msg:"tier"`
 }

--- a/cmd/tier-journal.go
+++ b/cmd/tier-journal.go
@@ -42,9 +42,9 @@ type tierJournal struct {
 }
 
 type jentry struct {
-	ObjName         string `msg:"obj"`
-	RemoteVersionID string `msg:"rvid"`
-	TierName        string `msg:"tier"`
+	ObjName   string `msg:"obj"`
+	VersionID string `msg:"vid"`
+	TierName  string `msg:"tier"`
 }
 
 const (
@@ -129,7 +129,7 @@ func (j *tierJournal) WalkEntries(fn walkFn) {
 			logger.LogIf(context.Background(), fmt.Errorf("tier-journal: failed to decode journal entry %s", err))
 			break
 		}
-		err = fn(entry.ObjName, entry.RemoteVersionID, entry.TierName)
+		err = fn(entry.ObjName, entry.VersionID, entry.TierName)
 		if err != nil && !isErrObjectNotFound(err) {
 			logger.LogIf(context.Background(), fmt.Errorf("tier-journal: failed to delete transitioned object %s from %s due to %s", entry.ObjName, entry.TierName, err))
 			// We add the entry into the active journal to try again

--- a/cmd/tier-journal_gen.go
+++ b/cmd/tier-journal_gen.go
@@ -30,6 +30,12 @@ func (z *jentry) DecodeMsg(dc *msgp.Reader) (err error) {
 				err = msgp.WrapError(err, "ObjName")
 				return
 			}
+		case "rvid":
+			z.RemoteVersionID, err = dc.ReadString()
+			if err != nil {
+				err = msgp.WrapError(err, "RemoteVersionID")
+				return
+			}
 		case "tier":
 			z.TierName, err = dc.ReadString()
 			if err != nil {
@@ -49,6 +55,153 @@ func (z *jentry) DecodeMsg(dc *msgp.Reader) (err error) {
 
 // EncodeMsg implements msgp.Encodable
 func (z jentry) EncodeMsg(en *msgp.Writer) (err error) {
+	// map header, size 3
+	// write "obj"
+	err = en.Append(0x83, 0xa3, 0x6f, 0x62, 0x6a)
+	if err != nil {
+		return
+	}
+	err = en.WriteString(z.ObjName)
+	if err != nil {
+		err = msgp.WrapError(err, "ObjName")
+		return
+	}
+	// write "rvid"
+	err = en.Append(0xa4, 0x72, 0x76, 0x69, 0x64)
+	if err != nil {
+		return
+	}
+	err = en.WriteString(z.RemoteVersionID)
+	if err != nil {
+		err = msgp.WrapError(err, "RemoteVersionID")
+		return
+	}
+	// write "tier"
+	err = en.Append(0xa4, 0x74, 0x69, 0x65, 0x72)
+	if err != nil {
+		return
+	}
+	err = en.WriteString(z.TierName)
+	if err != nil {
+		err = msgp.WrapError(err, "TierName")
+		return
+	}
+	return
+}
+
+// MarshalMsg implements msgp.Marshaler
+func (z jentry) MarshalMsg(b []byte) (o []byte, err error) {
+	o = msgp.Require(b, z.Msgsize())
+	// map header, size 3
+	// string "obj"
+	o = append(o, 0x83, 0xa3, 0x6f, 0x62, 0x6a)
+	o = msgp.AppendString(o, z.ObjName)
+	// string "rvid"
+	o = append(o, 0xa4, 0x72, 0x76, 0x69, 0x64)
+	o = msgp.AppendString(o, z.RemoteVersionID)
+	// string "tier"
+	o = append(o, 0xa4, 0x74, 0x69, 0x65, 0x72)
+	o = msgp.AppendString(o, z.TierName)
+	return
+}
+
+// UnmarshalMsg implements msgp.Unmarshaler
+func (z *jentry) UnmarshalMsg(bts []byte) (o []byte, err error) {
+	var field []byte
+	_ = field
+	var zb0001 uint32
+	zb0001, bts, err = msgp.ReadMapHeaderBytes(bts)
+	if err != nil {
+		err = msgp.WrapError(err)
+		return
+	}
+	for zb0001 > 0 {
+		zb0001--
+		field, bts, err = msgp.ReadMapKeyZC(bts)
+		if err != nil {
+			err = msgp.WrapError(err)
+			return
+		}
+		switch msgp.UnsafeString(field) {
+		case "obj":
+			z.ObjName, bts, err = msgp.ReadStringBytes(bts)
+			if err != nil {
+				err = msgp.WrapError(err, "ObjName")
+				return
+			}
+		case "rvid":
+			z.RemoteVersionID, bts, err = msgp.ReadStringBytes(bts)
+			if err != nil {
+				err = msgp.WrapError(err, "RemoteVersionID")
+				return
+			}
+		case "tier":
+			z.TierName, bts, err = msgp.ReadStringBytes(bts)
+			if err != nil {
+				err = msgp.WrapError(err, "TierName")
+				return
+			}
+		default:
+			bts, err = msgp.Skip(bts)
+			if err != nil {
+				err = msgp.WrapError(err)
+				return
+			}
+		}
+	}
+	o = bts
+	return
+}
+
+// Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
+func (z jentry) Msgsize() (s int) {
+	s = 1 + 4 + msgp.StringPrefixSize + len(z.ObjName) + 5 + msgp.StringPrefixSize + len(z.RemoteVersionID) + 5 + msgp.StringPrefixSize + len(z.TierName)
+	return
+}
+
+// DecodeMsg implements msgp.Decodable
+func (z *jentryV1) DecodeMsg(dc *msgp.Reader) (err error) {
+	var field []byte
+	_ = field
+	var zb0001 uint32
+	zb0001, err = dc.ReadMapHeader()
+	if err != nil {
+		err = msgp.WrapError(err)
+		return
+	}
+	for zb0001 > 0 {
+		zb0001--
+		field, err = dc.ReadMapKeyPtr()
+		if err != nil {
+			err = msgp.WrapError(err)
+			return
+		}
+		switch msgp.UnsafeString(field) {
+		case "obj":
+			z.ObjName, err = dc.ReadString()
+			if err != nil {
+				err = msgp.WrapError(err, "ObjName")
+				return
+			}
+		case "tier":
+			z.TierName, err = dc.ReadString()
+			if err != nil {
+				err = msgp.WrapError(err, "TierName")
+				return
+			}
+		default:
+			err = dc.Skip()
+			if err != nil {
+				err = msgp.WrapError(err)
+				return
+			}
+		}
+	}
+	return
+}
+
+// EncodeMsg implements msgp.Encodable
+func (z jentryV1) EncodeMsg(en *msgp.Writer) (err error) {
 	// map header, size 2
 	// write "obj"
 	err = en.Append(0x82, 0xa3, 0x6f, 0x62, 0x6a)
@@ -74,7 +227,7 @@ func (z jentry) EncodeMsg(en *msgp.Writer) (err error) {
 }
 
 // MarshalMsg implements msgp.Marshaler
-func (z jentry) MarshalMsg(b []byte) (o []byte, err error) {
+func (z jentryV1) MarshalMsg(b []byte) (o []byte, err error) {
 	o = msgp.Require(b, z.Msgsize())
 	// map header, size 2
 	// string "obj"
@@ -87,7 +240,7 @@ func (z jentry) MarshalMsg(b []byte) (o []byte, err error) {
 }
 
 // UnmarshalMsg implements msgp.Unmarshaler
-func (z *jentry) UnmarshalMsg(bts []byte) (o []byte, err error) {
+func (z *jentryV1) UnmarshalMsg(bts []byte) (o []byte, err error) {
 	var field []byte
 	_ = field
 	var zb0001 uint32
@@ -129,7 +282,7 @@ func (z *jentry) UnmarshalMsg(bts []byte) (o []byte, err error) {
 }
 
 // Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
-func (z jentry) Msgsize() (s int) {
+func (z jentryV1) Msgsize() (s int) {
 	s = 1 + 4 + msgp.StringPrefixSize + len(z.ObjName) + 5 + msgp.StringPrefixSize + len(z.TierName)
 	return
 }

--- a/cmd/tier-journal_gen.go
+++ b/cmd/tier-journal_gen.go
@@ -30,10 +30,10 @@ func (z *jentry) DecodeMsg(dc *msgp.Reader) (err error) {
 				err = msgp.WrapError(err, "ObjName")
 				return
 			}
-		case "rvid":
-			z.RemoteVersionID, err = dc.ReadString()
+		case "vid":
+			z.VersionID, err = dc.ReadString()
 			if err != nil {
-				err = msgp.WrapError(err, "RemoteVersionID")
+				err = msgp.WrapError(err, "VersionID")
 				return
 			}
 		case "tier":
@@ -66,14 +66,14 @@ func (z jentry) EncodeMsg(en *msgp.Writer) (err error) {
 		err = msgp.WrapError(err, "ObjName")
 		return
 	}
-	// write "rvid"
-	err = en.Append(0xa4, 0x72, 0x76, 0x69, 0x64)
+	// write "vid"
+	err = en.Append(0xa3, 0x76, 0x69, 0x64)
 	if err != nil {
 		return
 	}
-	err = en.WriteString(z.RemoteVersionID)
+	err = en.WriteString(z.VersionID)
 	if err != nil {
-		err = msgp.WrapError(err, "RemoteVersionID")
+		err = msgp.WrapError(err, "VersionID")
 		return
 	}
 	// write "tier"
@@ -96,9 +96,9 @@ func (z jentry) MarshalMsg(b []byte) (o []byte, err error) {
 	// string "obj"
 	o = append(o, 0x83, 0xa3, 0x6f, 0x62, 0x6a)
 	o = msgp.AppendString(o, z.ObjName)
-	// string "rvid"
-	o = append(o, 0xa4, 0x72, 0x76, 0x69, 0x64)
-	o = msgp.AppendString(o, z.RemoteVersionID)
+	// string "vid"
+	o = append(o, 0xa3, 0x76, 0x69, 0x64)
+	o = msgp.AppendString(o, z.VersionID)
 	// string "tier"
 	o = append(o, 0xa4, 0x74, 0x69, 0x65, 0x72)
 	o = msgp.AppendString(o, z.TierName)
@@ -129,10 +129,10 @@ func (z *jentry) UnmarshalMsg(bts []byte) (o []byte, err error) {
 				err = msgp.WrapError(err, "ObjName")
 				return
 			}
-		case "rvid":
-			z.RemoteVersionID, bts, err = msgp.ReadStringBytes(bts)
+		case "vid":
+			z.VersionID, bts, err = msgp.ReadStringBytes(bts)
 			if err != nil {
-				err = msgp.WrapError(err, "RemoteVersionID")
+				err = msgp.WrapError(err, "VersionID")
 				return
 			}
 		case "tier":
@@ -155,7 +155,7 @@ func (z *jentry) UnmarshalMsg(bts []byte) (o []byte, err error) {
 
 // Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
 func (z jentry) Msgsize() (s int) {
-	s = 1 + 4 + msgp.StringPrefixSize + len(z.ObjName) + 5 + msgp.StringPrefixSize + len(z.RemoteVersionID) + 5 + msgp.StringPrefixSize + len(z.TierName)
+	s = 1 + 4 + msgp.StringPrefixSize + len(z.ObjName) + 4 + msgp.StringPrefixSize + len(z.VersionID) + 5 + msgp.StringPrefixSize + len(z.TierName)
 	return
 }
 

--- a/cmd/tier-journal_gen_test.go
+++ b/cmd/tier-journal_gen_test.go
@@ -121,3 +121,116 @@ func BenchmarkDecodejentry(b *testing.B) {
 		}
 	}
 }
+
+func TestMarshalUnmarshaljentryV1(t *testing.T) {
+	v := jentryV1{}
+	bts, err := v.MarshalMsg(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	left, err := v.UnmarshalMsg(bts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(left) > 0 {
+		t.Errorf("%d bytes left over after UnmarshalMsg(): %q", len(left), left)
+	}
+
+	left, err = msgp.Skip(bts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(left) > 0 {
+		t.Errorf("%d bytes left over after Skip(): %q", len(left), left)
+	}
+}
+
+func BenchmarkMarshalMsgjentryV1(b *testing.B) {
+	v := jentryV1{}
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		v.MarshalMsg(nil)
+	}
+}
+
+func BenchmarkAppendMsgjentryV1(b *testing.B) {
+	v := jentryV1{}
+	bts := make([]byte, 0, v.Msgsize())
+	bts, _ = v.MarshalMsg(bts[0:0])
+	b.SetBytes(int64(len(bts)))
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		bts, _ = v.MarshalMsg(bts[0:0])
+	}
+}
+
+func BenchmarkUnmarshaljentryV1(b *testing.B) {
+	v := jentryV1{}
+	bts, _ := v.MarshalMsg(nil)
+	b.ReportAllocs()
+	b.SetBytes(int64(len(bts)))
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err := v.UnmarshalMsg(bts)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func TestEncodeDecodejentryV1(t *testing.T) {
+	v := jentryV1{}
+	var buf bytes.Buffer
+	msgp.Encode(&buf, &v)
+
+	m := v.Msgsize()
+	if buf.Len() > m {
+		t.Log("WARNING: TestEncodeDecodejentryV1 Msgsize() is inaccurate")
+	}
+
+	vn := jentryV1{}
+	err := msgp.Decode(&buf, &vn)
+	if err != nil {
+		t.Error(err)
+	}
+
+	buf.Reset()
+	msgp.Encode(&buf, &v)
+	err = msgp.NewReader(&buf).Skip()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func BenchmarkEncodejentryV1(b *testing.B) {
+	v := jentryV1{}
+	var buf bytes.Buffer
+	msgp.Encode(&buf, &v)
+	b.SetBytes(int64(buf.Len()))
+	en := msgp.NewWriter(msgp.Nowhere)
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		v.EncodeMsg(en)
+	}
+	en.Flush()
+}
+
+func BenchmarkDecodejentryV1(b *testing.B) {
+	v := jentryV1{}
+	var buf bytes.Buffer
+	msgp.Encode(&buf, &v)
+	b.SetBytes(int64(buf.Len()))
+	rd := msgp.NewEndlessReader(buf.Bytes(), b)
+	dc := msgp.NewReader(rd)
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		err := v.DecodeMsg(dc)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}

--- a/cmd/tier-journal_test.go
+++ b/cmd/tier-journal_test.go
@@ -1,0 +1,121 @@
+// Copyright (c) 2015-2021 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/tinylib/msgp/msgp"
+)
+
+// TestJEntryReadOldToNew1 - tests that adding the RemoteVersionID parameter to the
+// jentry struct does not cause unexpected errors when reading the serialized
+// old version into new version.
+func TestJEntryReadOldToNew1(t *testing.T) {
+	readOldToNewCases := []struct {
+		je  jentryV1
+		exp jentry
+	}{
+		{jentryV1{"obj1", "tier1"}, jentry{"obj1", "", "tier1"}},
+		{jentryV1{"obj1", ""}, jentry{"obj1", "", ""}},
+		{jentryV1{"", "tier1"}, jentry{"", "", "tier1"}},
+		{jentryV1{"", ""}, jentry{"", "", ""}},
+	}
+
+	var b bytes.Buffer
+	for _, item := range readOldToNewCases {
+		bs, err := item.je.MarshalMsg(nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		b.Write(bs)
+	}
+
+	mr := msgp.NewReader(&b)
+	for i, item := range readOldToNewCases {
+		var je jentry
+		err := je.DecodeMsg(mr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if je != item.exp {
+			t.Errorf("Case %d: Expected: %v Got: %v", i, item.exp, je)
+		}
+	}
+}
+
+// TestJEntryWriteNewToOldMix1 - tests that adding the RemoteVersionID parameter
+// to the jentry struct does not cause unexpected errors when writing. This
+// simulates the case when the active journal has entries in the older version
+// struct and due to errors new entries are added in the new version of the
+// struct.
+func TestJEntryWriteNewToOldMix1(t *testing.T) {
+	oldStructVals := []jentryV1{
+		{"obj1", "tier1"},
+		{"obj2", "tier2"},
+		{"obj3", "tier3"},
+	}
+	newStructVals := []jentry{
+		{"obj4", "", "tier1"},
+		{"obj5", "ver2", "tier2"},
+		{"obj6", "", "tier3"},
+	}
+
+	// Write old struct version values followed by new version values.
+	var b bytes.Buffer
+	for _, item := range oldStructVals {
+		bs, err := item.MarshalMsg(nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		b.Write(bs)
+	}
+	for _, item := range newStructVals {
+		bs, err := item.MarshalMsg(nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		b.Write(bs)
+	}
+
+	// Read into new struct version and check.
+	mr := msgp.NewReader(&b)
+	for i := 0; i < len(oldStructVals)+len(newStructVals); i++ {
+		var je jentry
+		err := je.DecodeMsg(mr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var expectedJe jentry
+		if i < len(oldStructVals) {
+			// For old struct values, the RemoteVersionID will be
+			// empty
+			expectedJe = jentry{
+				ObjName:         oldStructVals[i].ObjName,
+				RemoteVersionID: "",
+				TierName:        oldStructVals[i].TierName,
+			}
+		} else {
+			expectedJe = newStructVals[i-len(oldStructVals)]
+		}
+		if expectedJe != je {
+			t.Errorf("Case %d: Expected: %v, Got: %v", i, expectedJe, je)
+		}
+	}
+}

--- a/cmd/tier-journal_test.go
+++ b/cmd/tier-journal_test.go
@@ -107,9 +107,9 @@ func TestJEntryWriteNewToOldMix1(t *testing.T) {
 			// For old struct values, the RemoteVersionID will be
 			// empty
 			expectedJe = jentry{
-				ObjName:         oldStructVals[i].ObjName,
-				RemoteVersionID: "",
-				TierName:        oldStructVals[i].TierName,
+				ObjName:   oldStructVals[i].ObjName,
+				VersionID: "",
+				TierName:  oldStructVals[i].TierName,
 			}
 		} else {
 			expectedJe = newStructVals[i-len(oldStructVals)]

--- a/cmd/tier-sweeper.go
+++ b/cmd/tier-sweeper.go
@@ -41,15 +41,15 @@ import (
 //     logger.LogIf(ctx, err)
 // }
 type objSweeper struct {
-	Object                    string
-	Bucket                    string
-	ReqVersion                string // version ID set by application, applies only to DeleteObject and DeleteObjects APIs
-	Versioned                 bool
-	Suspended                 bool
-	TransitionStatus          string
-	TransitionTier            string
-	TransitionRemoteVersionID string
-	RemoteObject              string
+	Object              string
+	Bucket              string
+	ReqVersion          string // version ID set by application, applies only to DeleteObject and DeleteObjects APIs
+	Versioned           bool
+	Suspended           bool
+	TransitionStatus    string
+	TransitionTier      string
+	TransitionVersionID string
+	RemoteObject        string
 }
 
 // newObjSweeper returns an objSweeper for a given bucket and object.
@@ -117,7 +117,7 @@ func (os *objSweeper) SetTransitionState(info ObjectInfo) {
 	os.TransitionTier = info.TransitionTier
 	os.TransitionStatus = info.TransitionStatus
 	os.RemoteObject = info.transitionedObjName
-	os.TransitionRemoteVersionID = info.transitionRemoteVersionID
+	os.TransitionVersionID = info.transitionVersionID
 }
 
 // shouldRemoveRemoteObject determines if a transitioned object should be
@@ -145,9 +145,9 @@ func (os *objSweeper) shouldRemoveRemoteObject() (jentry, bool) {
 	}
 	if delTier {
 		return jentry{
-			ObjName:         os.RemoteObject,
-			RemoteVersionID: os.TransitionRemoteVersionID,
-			TierName:        os.TransitionTier,
+			ObjName:   os.RemoteObject,
+			VersionID: os.TransitionVersionID,
+			TierName:  os.TransitionTier,
 		}, true
 	}
 	return jentry{}, false

--- a/cmd/tier-sweeper.go
+++ b/cmd/tier-sweeper.go
@@ -41,14 +41,15 @@ import (
 //     logger.LogIf(ctx, err)
 // }
 type objSweeper struct {
-	Object           string
-	Bucket           string
-	ReqVersion       string // version ID set by application, applies only to DeleteObject and DeleteObjects APIs
-	Versioned        bool
-	Suspended        bool
-	TransitionStatus string
-	TransitionTier   string
-	RemoteObject     string
+	Object                    string
+	Bucket                    string
+	ReqVersion                string // version ID set by application, applies only to DeleteObject and DeleteObjects APIs
+	Versioned                 bool
+	Suspended                 bool
+	TransitionStatus          string
+	TransitionTier            string
+	TransitionRemoteVersionID string
+	RemoteObject              string
 }
 
 // newObjSweeper returns an objSweeper for a given bucket and object.
@@ -116,6 +117,7 @@ func (os *objSweeper) SetTransitionState(info ObjectInfo) {
 	os.TransitionTier = info.TransitionTier
 	os.TransitionStatus = info.TransitionStatus
 	os.RemoteObject = info.transitionedObjName
+	os.TransitionRemoteVersionID = info.transitionRemoteVersionID
 }
 
 // shouldRemoveRemoteObject determines if a transitioned object should be
@@ -142,7 +144,11 @@ func (os *objSweeper) shouldRemoveRemoteObject() (jentry, bool) {
 		delTier = true
 	}
 	if delTier {
-		return jentry{ObjName: os.RemoteObject, TierName: os.TransitionTier}, true
+		return jentry{
+			ObjName:         os.RemoteObject,
+			RemoteVersionID: os.TransitionRemoteVersionID,
+			TierName:        os.TransitionTier,
+		}, true
 	}
 	return jentry{}, false
 }

--- a/cmd/warm-backend-azure.go
+++ b/cmd/warm-backend-azure.go
@@ -53,19 +53,23 @@ func (az *warmBackendAzure) tier() azblob.AccessTierType {
 	}
 	return azblob.AccessTierType("")
 }
-func (az *warmBackendAzure) Put(ctx context.Context, object string, r io.Reader, length int64) error {
+
+// FIXME: add support for remote version ID in Azure remote tier and remove
+// this. Currently it's a no-op.
+
+func (az *warmBackendAzure) Put(ctx context.Context, object string, r io.Reader, length int64) (remoteVersionID, error) {
 	blobURL := az.serviceURL.NewContainerURL(az.Bucket).NewBlockBlobURL(az.getDest(object))
 	// set tier if specified -
 	if az.StorageClass != "" {
 		if _, err := blobURL.SetTier(ctx, az.tier(), azblob.LeaseAccessConditions{}); err != nil {
-			return azureToObjectError(err, az.Bucket, object)
+			return "", azureToObjectError(err, az.Bucket, object)
 		}
 	}
-	_, err := azblob.UploadStreamToBlockBlob(ctx, r, blobURL, azblob.UploadStreamToBlockBlobOptions{})
-	return azureToObjectError(err, az.Bucket, object)
+	res, err := azblob.UploadStreamToBlockBlob(ctx, r, blobURL, azblob.UploadStreamToBlockBlobOptions{})
+	return remoteVersionID(res.Version()), azureToObjectError(err, az.Bucket, object)
 }
 
-func (az *warmBackendAzure) Get(ctx context.Context, object string, opts WarmBackendGetOpts) (r io.ReadCloser, err error) {
+func (az *warmBackendAzure) Get(ctx context.Context, object string, rv remoteVersionID, opts WarmBackendGetOpts) (r io.ReadCloser, err error) {
 	if opts.startOffset < 0 {
 		return nil, InvalidRange{}
 	}
@@ -79,7 +83,7 @@ func (az *warmBackendAzure) Get(ctx context.Context, object string, opts WarmBac
 	return rc, nil
 }
 
-func (az *warmBackendAzure) Remove(ctx context.Context, object string) error {
+func (az *warmBackendAzure) Remove(ctx context.Context, object string, rv remoteVersionID) error {
 	blob := az.serviceURL.NewContainerURL(az.Bucket).NewBlobURL(az.getDest(object))
 	_, err := blob.Delete(ctx, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
 	return azureToObjectError(err, az.Bucket, object)

--- a/cmd/warm-backend-gcs.go
+++ b/cmd/warm-backend-gcs.go
@@ -43,7 +43,11 @@ func (gcs *warmBackendGCS) getDest(object string) string {
 	}
 	return destObj
 }
-func (gcs *warmBackendGCS) Put(ctx context.Context, key string, data io.Reader, length int64) error {
+
+// FIXME: add support for remote version ID in GCS remote tier and remove this.
+// Currently it's a no-op.
+
+func (gcs *warmBackendGCS) Put(ctx context.Context, key string, data io.Reader, length int64) (remoteVersionID, error) {
 	object := gcs.client.Bucket(gcs.Bucket).Object(gcs.getDest(key))
 	//TODO: set storage class
 	w := object.NewWriter(ctx)
@@ -51,13 +55,13 @@ func (gcs *warmBackendGCS) Put(ctx context.Context, key string, data io.Reader, 
 		w.ObjectAttrs.StorageClass = gcs.StorageClass
 	}
 	if _, err := io.Copy(w, data); err != nil {
-		return gcsToObjectError(err, gcs.Bucket, key)
+		return "", gcsToObjectError(err, gcs.Bucket, key)
 	}
 
-	return w.Close()
+	return "", w.Close()
 }
 
-func (gcs *warmBackendGCS) Get(ctx context.Context, key string, opts WarmBackendGetOpts) (r io.ReadCloser, err error) {
+func (gcs *warmBackendGCS) Get(ctx context.Context, key string, rv remoteVersionID, opts WarmBackendGetOpts) (r io.ReadCloser, err error) {
 	// GCS storage decompresses a gzipped object by default and returns the data.
 	// Refer to https://cloud.google.com/storage/docs/transcoding#decompressive_transcoding
 	// Need to set `Accept-Encoding` header to `gzip` when issuing a GetObject call, to be able
@@ -73,7 +77,7 @@ func (gcs *warmBackendGCS) Get(ctx context.Context, key string, opts WarmBackend
 	return r, nil
 }
 
-func (gcs *warmBackendGCS) Remove(ctx context.Context, key string) error {
+func (gcs *warmBackendGCS) Remove(ctx context.Context, key string, rv remoteVersionID) error {
 	err := gcs.client.Bucket(gcs.Bucket).Object(gcs.getDest(key)).Delete(ctx)
 	return gcsToObjectError(err, gcs.Bucket, key)
 }

--- a/cmd/warm-backend-s3.go
+++ b/cmd/warm-backend-s3.go
@@ -56,14 +56,17 @@ func (s3 *warmBackendS3) getDest(object string) string {
 	return destObj
 }
 
-func (s3 *warmBackendS3) Put(ctx context.Context, object string, r io.Reader, length int64) error {
-	_, err := s3.client.PutObject(ctx, s3.Bucket, s3.getDest(object), r, length, minio.PutObjectOptions{StorageClass: s3.StorageClass})
-	return s3.ToObjectError(err, object)
+func (s3 *warmBackendS3) Put(ctx context.Context, object string, r io.Reader, length int64) (remoteVersionID, error) {
+	res, err := s3.client.PutObject(ctx, s3.Bucket, s3.getDest(object), r, length, minio.PutObjectOptions{StorageClass: s3.StorageClass})
+	return remoteVersionID(res.VersionID), s3.ToObjectError(err, object)
 }
 
-func (s3 *warmBackendS3) Get(ctx context.Context, object string, opts WarmBackendGetOpts) (io.ReadCloser, error) {
+func (s3 *warmBackendS3) Get(ctx context.Context, object string, rv remoteVersionID, opts WarmBackendGetOpts) (io.ReadCloser, error) {
 	gopts := minio.GetObjectOptions{}
 
+	if rv != "" {
+		gopts.VersionID = string(rv)
+	}
 	if opts.startOffset >= 0 && opts.length > 0 {
 		if err := gopts.SetRange(opts.startOffset, opts.startOffset+opts.length-1); err != nil {
 			return nil, s3.ToObjectError(err, object)
@@ -78,8 +81,12 @@ func (s3 *warmBackendS3) Get(ctx context.Context, object string, opts WarmBacken
 	return r, nil
 }
 
-func (s3 *warmBackendS3) Remove(ctx context.Context, object string) error {
-	err := s3.client.RemoveObject(ctx, s3.Bucket, s3.getDest(object), minio.RemoveObjectOptions{})
+func (s3 *warmBackendS3) Remove(ctx context.Context, object string, rv remoteVersionID) error {
+	ropts := minio.RemoveObjectOptions{}
+	if rv != "" {
+		ropts.VersionID = string(rv)
+	}
+	err := s3.client.RemoveObject(ctx, s3.Bucket, s3.getDest(object), ropts)
 	return s3.ToObjectError(err, object)
 }
 

--- a/cmd/warm-backend.go
+++ b/cmd/warm-backend.go
@@ -36,9 +36,9 @@ type WarmBackendGetOpts struct {
 
 // WarmBackend provides interface to be implemented by remote tier backends
 type WarmBackend interface {
-	Put(ctx context.Context, object string, r io.Reader, length int64) error
-	Get(ctx context.Context, object string, opts WarmBackendGetOpts) (io.ReadCloser, error)
-	Remove(ctx context.Context, object string) error
+	Put(ctx context.Context, object string, r io.Reader, length int64) (remoteVersionID, error)
+	Get(ctx context.Context, object string, rv remoteVersionID, opts WarmBackendGetOpts) (io.ReadCloser, error)
+	Remove(ctx context.Context, object string, rv remoteVersionID) error
 	InUse(ctx context.Context) (bool, error)
 }
 
@@ -48,7 +48,7 @@ const probeObject = "probeobject"
 // to perform all operations defined in the WarmBackend interface.
 func checkWarmBackend(ctx context.Context, w WarmBackend) error {
 	var empty bytes.Reader
-	err := w.Put(ctx, probeObject, &empty, 0)
+	rv, err := w.Put(ctx, probeObject, &empty, 0)
 	if err != nil {
 		return tierPermErr{
 			Op:  tierPut,
@@ -56,7 +56,7 @@ func checkWarmBackend(ctx context.Context, w WarmBackend) error {
 		}
 	}
 
-	_, err = w.Get(ctx, probeObject, WarmBackendGetOpts{})
+	_, err = w.Get(ctx, probeObject, rv, WarmBackendGetOpts{})
 	if err != nil {
 		switch {
 		case isErrBucketNotFound(err):
@@ -71,7 +71,7 @@ func checkWarmBackend(ctx context.Context, w WarmBackend) error {
 		}
 	}
 
-	if err = w.Remove(ctx, probeObject); err != nil {
+	if err = w.Remove(ctx, probeObject, rv); err != nil {
 		return tierPermErr{
 			Op:  tierDelete,
 			Err: err,
@@ -114,6 +114,10 @@ func errIsTierPermError(err error) bool {
 	var tpErr tierPermErr
 	return errors.As(err, &tpErr)
 }
+
+// remoteVersionID represents the version id of an object in the remote tier.
+// Its usage is remote tier cloud implementation specific.
+type remoteVersionID string
 
 // newWarmBackend instantiates the tier type specific WarmBackend, runs
 // checkWarmBackend on it.

--- a/cmd/xl-storage-format-v2.go
+++ b/cmd/xl-storage-format-v2.go
@@ -875,6 +875,9 @@ func (z *xlMetaV2) AddVersion(fi FileInfo) error {
 		if fi.TransitionedObjName != "" {
 			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedObjectName] = []byte(fi.TransitionedObjName)
 		}
+		if fi.TransitionRemoteVersionID != "" {
+			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedRemoteVersionID] = []byte(fi.TransitionRemoteVersionID)
+		}
 		if fi.TransitionTier != "" {
 			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionTier] = []byte(fi.TransitionTier)
 		}
@@ -1019,6 +1022,9 @@ func (j xlMetaV2Object) ToFileInfo(volume, path string) (FileInfo, error) {
 	}
 	if o, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionedObjectName]; ok {
 		fi.TransitionedObjName = string(o)
+	}
+	if rv, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionedRemoteVersionID]; ok {
+		fi.TransitionRemoteVersionID = string(rv)
 	}
 	if sc, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionTier]; ok {
 		fi.TransitionTier = string(sc)
@@ -1188,6 +1194,7 @@ func (z *xlMetaV2) DeleteVersion(fi FileInfo) (string, bool, error) {
 				case fi.TransitionStatus == lifecycle.TransitionComplete:
 					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionStatus] = []byte(fi.TransitionStatus)
 					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedObjectName] = []byte(fi.TransitionedObjName)
+					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedRemoteVersionID] = []byte(fi.TransitionRemoteVersionID)
 					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionTier] = []byte(fi.TransitionTier)
 				default:
 					z.Versions = append(z.Versions[:i], z.Versions[i+1:]...)

--- a/cmd/xl-storage-format-v2.go
+++ b/cmd/xl-storage-format-v2.go
@@ -875,8 +875,8 @@ func (z *xlMetaV2) AddVersion(fi FileInfo) error {
 		if fi.TransitionedObjName != "" {
 			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedObjectName] = []byte(fi.TransitionedObjName)
 		}
-		if fi.TransitionRemoteVersionID != "" {
-			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedRemoteVersionID] = []byte(fi.TransitionRemoteVersionID)
+		if fi.TransitionVersionID != "" {
+			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedVersionID] = []byte(fi.TransitionVersionID)
 		}
 		if fi.TransitionTier != "" {
 			ventry.ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionTier] = []byte(fi.TransitionTier)
@@ -1023,8 +1023,8 @@ func (j xlMetaV2Object) ToFileInfo(volume, path string) (FileInfo, error) {
 	if o, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionedObjectName]; ok {
 		fi.TransitionedObjName = string(o)
 	}
-	if rv, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionedRemoteVersionID]; ok {
-		fi.TransitionRemoteVersionID = string(rv)
+	if rv, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionedVersionID]; ok {
+		fi.TransitionVersionID = string(rv)
 	}
 	if sc, ok := j.MetaSys[ReservedMetadataPrefixLower+TransitionTier]; ok {
 		fi.TransitionTier = string(sc)
@@ -1194,7 +1194,7 @@ func (z *xlMetaV2) DeleteVersion(fi FileInfo) (string, bool, error) {
 				case fi.TransitionStatus == lifecycle.TransitionComplete:
 					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionStatus] = []byte(fi.TransitionStatus)
 					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedObjectName] = []byte(fi.TransitionedObjName)
-					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedRemoteVersionID] = []byte(fi.TransitionRemoteVersionID)
+					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionedVersionID] = []byte(fi.TransitionVersionID)
 					z.Versions[i].ObjectV2.MetaSys[ReservedMetadataPrefixLower+TransitionTier] = []byte(fi.TransitionTier)
 				default:
 					z.Versions = append(z.Versions[:i], z.Versions[i+1:]...)

